### PR TITLE
Use non-deprecated etcd library; vendorize

### PR DESCRIPTION
**Pull request by tschuy**
It's that time again, we're switching to vendoring in another thing!

I can throw out the vendoring, but it seems like a nice thing to have anyway.

I also switched to the official etcd client library.

### CODE REVIEW

**Comment by tschuy** [comment]
@brianredbeard

**Comment by tschuy** [comment]
There's still a large amount of `third_party.go` dep management that I have yet to rework, that blocks this. But I'd like to get this checked out before I continue trudging through that just to make sure I'm doing work we actually want.

**Comment by tschuy** [comment]
could you also verify that everything in `vendor/` exists properly? the `added` number should be a lot higher than 97, and I just want to make sure that's just github bailing for the size of the commit.

**Comment by brianredbeard** [comment]
Likely it's just a presentation bug on the GitHub side:

```
$ git show --stat | grep "^ vendor/" | wc -l 
143
```

File: handlers/new.go [hunk 1]
```diff
@@ -11,10 +11,19 @@ import (
...
 	"strconv"
+	"time"
 
-	"github.com/coreos/go-etcd/etcd"
+	"github.com/coreos/etcd/client"
+	"golang.org/x/net/context"
 )
 
+var cfg = client.Config{
+	Endpoints: []string{"http://127.0.0.1:2379"},
```
Comment (brianredbeard) [inline handlers/new.go]: Looks good, but we just need to flag this to make it configurable later.

**Comment by brianredbeard** [comment]
Assuming that count is good, we should be ready to merge so we can keep plowing on.

File: handlers/new.go (hunk 1 above)
Comment (tschuy) [inline handlers/new.go]: the goal of this pr was moreso to duplicate existing behavior with the new library. I'd like to do this in another PR.

... also because #42 depends on this.

**Comment by brianredbeard** [comment]
LGTM
Let's do this
